src/uri/params.cr

```diff
@@ -94,21 +94,55 @@ class URI
       end
     end
 
+    # Appends the given key value pairs as a url-encoded URI form/query to the given `io`.
+    #
+    # ```
+    # require "uri/params"
+    #
+    # io = IO::Memory.new
+    # URI::Params.encode({"foo" => "bar", "baz" => ["quux", "quuz"]}, io)
+    # io.to_s # => "foo=bar&baz=quux&baz=quuz"
+    # ```
+    def self.encode(hash : Hash(String, String | Array(String)), io : IO) : Nil
```

Comment:
Based on https://github.com/crystal-lang/crystal/issues/5916, we may want to consider making the `io` parameter first.

Reply:
Ah. Yeah, I was a little confused on that with some put it first and some second :joy:  I'll swap those.